metaseq/data/streaming_src_tgt_dataset.py

```diff
@@ -164,3 +165,45 @@ def yield_src_tgt_blocks(iterable, block_size, drop_last, padding_idx):
             "src_block": src_block,
             "tgt_block": tgt_block,
         }
+
+
+def yield_src_tgt_single_sentences_pad_8(iterable, block_size, drop_last, padding_idx):
```

Comment:
Is there a specific reason why it has to be `8` in particular? Can this be generalized to any value (or will that never be needed)?

Reply:
8 worked well across the most number of models and did not trigger the internal assert from megatron. Once document attention is debugged and landed, we can use that instead of this and this will just become a fallback mechanism in case document attention gives errors.